pkg/internal/infrastructure/templates/main.tpl.tf

```
@@ -102,11 +103,18 @@ resource "openstack_networking_subnet_v2" "cluster" {
   dns_nameservers = []
   {{- end }}
 }
+{{ else -}}
+data "openstack_networking_subnet_v2" "cluster" {
+  subnet_id = "{{ .networks.subnet }}"
+}
+{{- end }}
 
+{{- if (or (.create.router) (.create.subnet)) }}
```

```suggestion
{{ if (or (.create.router) (.create.subnet)) }}
```
is the dash here necessary ?